pkg/cmd/pulumi/env.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2016-2022, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// A small library for creating consistent and documented environmental variable accesses.
+//
+// Public environmental variables should be declared as a module level variable.
+
+package main
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/common/diag"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/env"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
+	declared "github.com/pulumi/pulumi/sdk/v3/go/common/util/env"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/result"
+)
+
+func newEnvCmd() *cobra.Command {
+	return &cobra.Command{
+		Use:   "env",
+		Short: "An overview of the environmental variables used by pulumi",
+		Args:  cmdutil.NoArgs,
+		// Since most variables won't be included here, we hide the command. We will
+		// unhide once most existing variables are using the new env var framework and
+		// show up here.
+		Hidden: !env.Experimental.Value(),
+		Run: cmdutil.RunResultFunc(func(cmd *cobra.Command, args []string) result.Result {
+			table := cmdutil.Table{
+				Headers: []string{"Variable", "Description", "Value"},
+			}
+			var foundError bool
```

Comment:
We don't "fail fast" for invalid environmental variables right now, but we trivially could. I think adding env var "fail fast" should be a separate PR, if we add it, since it would be a breaking change.